### PR TITLE
Add Makefile rule to detect stale vendor directory

### PR DESCRIPTION
**Pull request by dgrisonnet**
Create verify-vendor Makefile rule and add it to the verify rule in
order to detect stale vendor directory. Prior to this change, the verify
rule was vendoring the project without actually verifying that changes
needed to be pushed.

I also reordered the existing rules to be grouped.

/cc @s-urbaniak

### CODE REVIEW

**Comment by serathius** [comment]
Would it not be simpler just to remove vendor and depend on go.mod?

**Comment by s-urbaniak** [comment]
agreed with @serathius. This seems to be the general way forward anyways.

**Comment by dgrisonnet** [comment]
@serathius I applied your comment, PTAL :slightly_smiling_face:

File: Makefile [hunk 1]
```diff
@@ -8,29 +8,44 @@ OPENAPI_PATH=./vendor/k8s.io/kube-openapi
...
 verify-gofmt:
 	./hack/gofmt-all.sh -v
 
-gofmt:
-	./hack/gofmt-all.sh
+.PHONY: verify
+verify: verify-vendor verify-gofmt
+
+.PHONY: verify-vendor
+verify-vendor: go.mod
```
Comment (serathius) [inline Makefile:39]: Any reason to keep `vendor` directory? Is adding this an in between step needed before removal?

I'm not golang expert so maybe I'm missing something, but if vendor can be derived from `go.mod` and `go.sum` what is the benefit of keeping `vendor` directory in code?

**Pull request closed without merging**